common/src/main/java/com/wynntils/wc/utils/scoreboard/ScoreboardManager.java

```diff
@@ -0,0 +1,178 @@
+            changeType.ifPresent(change -> {
+                changeTypes.putIfAbsent(change, new HashSet<>());
+                changeTypes.get(change).add(new WynntilsScoreboardUpdateEvent.Change(line.line(), processed.method()));
+            });
+        }
+
+        reconstructedScoreboard.sort(
+                Comparator.comparingInt(ScoreboardLine::index).reversed());
+
+        WynntilsMod.getEventBus().post(new WynntilsScoreboardUpdateEvent(changeTypes));
+    };
+
+    private static Optional<WynntilsScoreboardUpdateEvent.ChangeType> getChangeType(ScoreboardLineChange change) {
+        String line = change.lineText();
+        String withoutFormat = ChatFormatting.stripFormatting(line);
+
+        if (withoutFormat == null) {
+            return Optional.empty();
+        }
+
+        if (GUILD_ATTACK_TERRITORY_PATTERN.matcher(line).matches() || GUILD_ATTACK_UPCOMING.equals(line)) {
+            return Optional.of(WynntilsScoreboardUpdateEvent.ChangeType.GuildAttackTimer);
+        }
+
+        if (PARTY_PLAYER_HEALTH_PATTERN.matcher(withoutFormat).matches()
+                || PARTY_PATTERN.matcher(line).matches()) {
+            return Optional.of(WynntilsScoreboardUpdateEvent.ChangeType.Party);
+        }
+
+        if (ObjectiveManager.OBJECTIVE_PATTERN.matcher(line).matches()) {
+            return Optional.of(WynntilsScoreboardUpdateEvent.ChangeType.Objective);
+        }
+
+        if (withoutFormat.matches("À+")
+                || OBJECTIVE_HEADER_PATTERN.matcher(withoutFormat).matches()
+                || GUILD_OBJECTIVE_HEADER_PATTERN.matcher(withoutFormat).matches()) {
+            return Optional.empty();
+        }
+
+        // For unidentified text, assume it is a quest description
+        return Optional.of(WynntilsScoreboardUpdateEvent.ChangeType.Quest);
+    }
+
+    public static void init() {
+        WynntilsMod.getEventBus().register(ScoreboardManager.class);
+        WynntilsMod.getEventBus().register(ObjectiveManager.class);
+        WynntilsMod.getEventBus().register(QuestManager.class);
+    }
+
+    @SubscribeEvent
+    public static void onSetScore(ScoreboardSetScoreEvent event) {
+        if (trackedObjectivePlayer == null) {
+            getTrackedObjectivePlayer();
+
+            if (trackedObjectivePlayer == null) {
+                return;
+            }
+        }
+
+        // Prevent duplication
+        if (!Objects.equals(event.getObjectiveName(), trackedObjectivePlayer.getName())) {
+            return;
+        }
+
+        queuedChanges.add(new ScoreboardLineChange(event.getOwner(), event.getMethod(), event.getScore()));
+    }
+
+    private static void getTrackedObjectivePlayer() {
+        Scoreboard scoreboard = McUtils.player().getScoreboard();
+
+        List<Objective> objectives = scoreboard.getObjectives().stream().toList();
+
+        if (objectives.isEmpty()) {
+            return;
+        }
+
+        trackedObjectivePlayer = objectives.get(0);
```

Comment:
Are you feeling confident that this is OK, just grabbing the first? I remember I wrote some code in legacy trying to find a player named "sb_<playername>" to track the scoreboard changes. Iirc, there also was a "fb_<playername>" but it did not get all scoreboard updates. Or maybe not. It's all a bit unclear this much afterwards.

Reply:
IMO, let's assume it is correct and we will fix it if it is not. I don't want to over engineer it if it is not needed.